SI/Sem.05/Pract.05/Task02.cpp

```diff
@@ -0,0 +1,86 @@
+// Task02.cpp : This file contains the 'main' function. Program execution begins and ends there.
+//
+
+#include <iostream>
+void SortEvensOdds(int* numbers, int length);
+void InsertElement(int indexElement, int indexToInsert, int* numbers, int length);
+void SortArray(int* numbers, int length);
+int main()
+{
+	//int numbers[] = { 1,88,2,3,5,11,12,6,7,4 };
+	int numbers[] = { 5,2,88,23,45,45,1,8,43,0 };
+	//int numbers[] = { 3,5,7,9,11,7,5,5,5,5 };
+	//int numbers[] = { 2,4,6,66,2,-4,6,8,10,2 };
+
+	int length = 10;
+	SortEvensOdds(numbers, length);
+	SortArray(numbers, length);
+	for (int i = 0; i < length; i++)
+	{
+		if (i == length-1)
+		{
+			std::cout << numbers[i]<<std::endl;
+		}
+		else
+		{
+			std::cout << numbers[i] << ", ";
+		}
+	}
+}
+
+void SortEvensOdds(int* numbers, int length)
+{
+	int counterInsertedElements = 0;
+	for (int i = 0; i < length - 1 - counterInsertedElements; i++)
+	{
+		int element = numbers[i];
+		if (element % 2 != 0)
+		{
+			InsertElement(i, length - 1 - counterInsertedElements, numbers, length);
+			i--;
+			counterInsertedElements++;
+		}
+	}
+}
```

Comment:
Good job for this code, but i < length - counter should be your check in your for loop condition, is there something that I am missing?